### DISCUSSION TITLE
Indeterminate progress indicator (Loading spinner)

### DISCUSSION THREAD

**Discussion by kelliedesigner** [ministryofjustice/moj-frontend#237] [EXPERIMENTAL: Components, pages and patterns]
# Existing

### Existing GitHub discussions

- [Indeterminate progress indicator](https://github.com/alphagov/govuk-design-system-backlog/issues/164) - in the GOV.UK community backlog
- [Loading spinner](https://github.com/alphagov/govuk-design-system-backlog/issues/28) - in the GOV.UK community backlog
- [Loading spinner](https://github.com/nhsuk/nhsuk-service-manual-community-backlog/issues/255) - In the NHS community backlog

### Existing components

- [Loading spinner](https://github.com/alphagov/govuk-design-system-backlog/issues/28#issuecomment-489061680) - Malcolm from LAA shared in GitHub

# Work in MOJ

**Product name** - Digital Prison Service
![image](https://user-images.githubusercontent.com/6122118/126620407-6df14ec7-fabb-43e8-b2f1-f6a325feb026.png)

There was a need because sometimes user have to wait was sometimes up to 15 seconds due to the amount of data.

**Comment by robertjmccarthy** [comment]
@acsauk has contributed a loading spinner modal to the design system. As this discussion already exists, I am adding the contribution here:

### Contribution purpose
We needed a way to show a user that a long running task (in this case a virus scan on uploads) was in progress along with providing progress updates. The solution uses a modal to prevent the user from trying to upload further files during the upload progress and updates 'x of x files uploaded' text as the files are scanned.

### Examples
- https://github.com/ministryofjustice/opg-modernising-lpa/blob/main/web/assets/file-upload-modal.js
- https://github.com/ministryofjustice/opg-modernising-lpa/blob/main/web/assets/scss/patterns/_loading_spinner.scss
- https://github.com/ministryofjustice/opg-modernising-lpa/blob/main/web/template/donor/upload_evidence.gohtml
